src/Krizalys/Onedrive/Client.php

```diff
@@ -618,7 +619,7 @@ public function createFile($name, $parentId = null, $content = '')
 
         // TODO: some versions of cURL cannot PUT memory streams? See here for a
         // workaround: https://bugs.php.net/bug.php?id=43468
-        $file = $this->apiPut($parentId . '/files/' . urlencode($name), $stream);
+        $file = $this->apiPut($parentId . '/files/' . urlencode($name) . '?overwrite=' . ($overwrite ? 'true' : 'ChooseNewName'), $stream);
```

Comment:
I did more testing, and I removed my previous comment after having discovered the following:

* passing `overwrite=true` (or not passing anything at all) will overwrite the file if a file with this name already exists (current behavior)
* passing `overwrite=false` will throw an Exception if a file with this name already exists
* passing `overwrite=ChooseNewName` will create a new file if a file with this name already exists (the file will be named after the original one, followed by a sequence number, such as `File 1` if `File`.

It would be cool to support these 3 policies. Would you like to add support for the `overwrite=false` case and perhaps passing constants such as `OVERWRITE_ALWAYS`, `OVERWRITE_NEVER`, `OVERWRITE_RENAME` when calling `createFile`?